src/main/java/com/sergiuszg/githubdetailsservice/configuration/DefaultErrorDecoder.java

```diff
@@ -0,0 +1,31 @@
+package com.sergiuszg.githubdetailsservice.configuration;
+
+import com.sergiuszg.githubdetailsservice.exceptions.ResourceNotFoundException;
+import feign.FeignException;
+import feign.Response;
+import feign.RetryableException;
+import feign.codec.ErrorDecoder;
+import org.springframework.stereotype.Component;
+
+@Component
+public class DefaultErrorDecoder implements ErrorDecoder {
+
+    @Override
+    public Exception decode(String methodKey, Response response) {
+        FeignException exception = feign.FeignException.errorStatus(methodKey, response);
+        int status = response.status();
+        if (status >= 500) {
+            return new RetryableException(
+                    response.status(),
+                    exception.getMessage(),
+                    response.request().httpMethod(),
+                    exception,
+                    null,
+                    response.request());
+        }
+        if (status == 404) {
+            return new ResourceNotFoundException("Resource not found");
```

Comment:
Moznaby sie zastanowic czy my tutaj nie chcemy tego usunac poniewaz jest to bardzo ogolny message i nie za duzo mowi uzytkownikowi.